src/coreclr/jit/morph.cpp

```diff
@@ -8358,10 +8358,20 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
             return tree;
         }
 
+#ifdef DEBUG
```

Comment:
As I noted over in #107542 we don't want to have control flow logic conditional on `DEBUG` like this...  we need to find a better fix.